flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/ExecutionVertexOperations.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.runtime.scheduler;
+
+import org.apache.flink.runtime.JobException;
+import org.apache.flink.runtime.executiongraph.ExecutionVertex;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Operations on the ExecutionVertex.
+ */
+interface ExecutionVertexOperations {
+
+	void deploy(ExecutionVertex executionVertex) throws JobException;
```

Comment:
ExecutionVertexID as the param should be a better for interfaces.

Reply:
That would require `DefaultExecutionVertexOperations` to have a reference to `ExecutionGraph` which is also not nice.

Reply:
A `ExecutionVertexID -> ExecutionVertex` function which references `SchedulerBase#getExecutionVertex` would be enough rather than the entire `ExecutionGraph`. 
But that requires the `DefaultScheduler` to be responsible for creating `DefaultExecutionVertexOperations`, which makes it harder to test the scheduler with `FailingExecutionVertexOperationsDecorator`.

One solution I can come up with is to use factory to create **ExecutionVertexOperations**. I did a try in [this commit](https://github.com/zhuzhurk/flink/commit/21b09575fb5818c76313e6a73c8a826626e1bc0a).
However the changes was more than I had supposed. So you can decide whether we should have such changes.

Reply:
I'd like not to do this change now because I am sure that `ExecutionVertexOperations` will undergo more refactorings in the future.

Reply:
Ok.